pkg/generators/openapi_test.go

```diff
@@ -495,3 +495,62 @@ Dependencies: []string{
 },
 `, buffer.String())
 }
+
+func TestExtensions(t *testing.T) {
+	err, assert, buffer := testOpenAPITypeWritter(t, `
+package foo
+
+// Blah is a test.
+// +k8s:openapi-gen=true
+// +k8s:openapi-gen=x-kubernetes-type-tag:type_test
+type Blah struct {
+	// a member with a list type
+	// +listType=map
+	// +listTypeKey=port
+	// +listTypeKey=protocol
+	WithListField []string
+}
+		`)
+	if err != nil {
+		t.Fatal(err)
+	}
+	assert.Equal(`"base/foo.Blah": {
+Schema: spec.Schema{
```

Comment:
This is really hard to read. Can we print prettified json instead?

Reply:
Lol, this is not json, is it. It's not hard to programmatically invoke gofmt, I think. I guess that can be another PR.

Reply:
I'll address this formatting issue (for the previous tests as well) in a future PR.

Reply:
the generated code is not gofmt compliant and the scripts will run gofmt on them. if you format this string then you need to gofmt the output of generator (which is dynamic) too before comparing it with assert.Equal. I would say this is just too much to do for tests but feel free to find a solution for it if you want to spend time on it.